test-case/check-alsabat.sh

```diff
@@ -50,39 +56,56 @@ then
 	exit 2
 fi
 
+pcmid_p=${pcm_p:-1}
+
 [[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
 
-function __upload_wav_file
-{
-    # upload the alsabat wav file
-    for file in /tmp/bat.wav.*
-    do
-	    size=`ls -l $file | awk '{print $5}'`
-	    if [[ $size -gt 0 ]]; then
-		    cp $file $LOG_ROOT/
-	    fi
-    done
-}
+func_pipeline_export "$tplg" "type:playback & id:$pcmid_p"
+
+# parser the parameters of the specified playback pipeline
+channel=$(func_pipeline_parse_value 0 ch_max)
+rate=$(func_pipeline_parse_value 0 rate)
+fmts=$(func_pipeline_parse_value 0 fmts)
 
 # check the PCMs before alsabat test
 dlogi "check the PCMs before alsabat test"
-[[ $(aplay -Dplug$pcm_p -d 1 /dev/zero -q) ]] && die "Failed to play on PCM: $pcm_p"
-[[ $(arecord -Dplug$pcm_c -d 1 /dev/null -q) ]] && die "Failed to capture on PCM: $pcm_c"
+aplay -Dplug"$pcm_p" -d 1 /dev/zero -q || die "Failed to play on PCM: $pcm_p"
+arecord -Dplug"$pcm_c" -d 1 /dev/null -q || die "Failed to capture on PCM: $pcm_c"
 
 # alsabat test
-# different PCMs may support different audio formats(like samplerate, channel-counting, etc.).
-# use plughw to do the audio format conversions. So we don't need to specify them for each PCM.
-dlogc "alsabat -Pplug$pcm_p --standalone -n $frames -F $frequency"
-alsabat -Pplug$pcm_p --standalone -n $frames -F $frequency &
-# playback may have low latency, add one second delay to aviod recording zero at beginning.
-sleep 1
-dlogc "alsabat -Cplug$pcm_c -F $frequency"
-alsabat -Cplug$pcm_c -F $frequency
-
-# upload failed wav file
-if [[ $? != 0 ]]; then
-	__upload_wav_file
-	exit 1
-fi
+for format in $fmts
+    do
+        if [ "$format" == "S24_LE" ]; then
+            dlogi "S24_LE is not supported, skip to test this format"
+            continue
+        fi
+
+        dlogc "alsabat -P $pcm_p --standalone -n $frames -F $frequency -c $channel -r $rate -f $format"
+        timeout -k 2 6 alsabat -P "$pcm_p" --standalone -n "$frames" -F "$frequency" -c "$channel" \
```

Comment:
what does 2 6 mean? seriously add comments on what this does.

Reply:
The meanings of 2 and 6 are at the top of the `man timeout` page. I don't think we need to copy/paste into this script the start of the man page of every unusual command but maybe we can have the  `--long` options like this:
```
timeout --kill-after 2 6 alsabat ...
```